### PR TITLE
Check that the event source is enabled before dumping config

### PR DESCRIPTION
**Pull request by davidfowl**
- Make sure the event source is enabled before dumping configuration.
- Change the level to log always

Follow up to https://github.com/dotnet/aspnetcore/pull/34680

### CODE REVIEW

**Review by noahfalk — APPROVED** [review]
LGTM : )

File: src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs [hunk 1]
```diff
@@ -220,7 +220,7 @@ public void TlsHandshakeFailed(string connectionId)
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        [Event(11, Level = EventLevel.Informational)]
+        [Event(11, Level = EventLevel.LogAlways)]
```
Comment (shirhatti) [inline src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs:223]: Why `LogAlways`?

Comment (davidfowl) [reply src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs:223]: Payload is small enough to not cause any damage

**Comment by captainsafia** [comment]
Using my admin privileges to merge. Builds on this PR have been failing due to unrelated flakiness with the Component tests. Merging to unblock.

**Pull request merged**